### PR TITLE
Remove deadcode: Interface TokenExchanger, and only implementation (Google STS)

### PR DESCRIPTION
**Pull request by howardjohn**
Followup from https://github.com/istio/istio/pull/50989

### CODE REVIEW

**Review by kyessenov — COMMENTED** [review]
Mostly LG. We haven't absorbed the CA deletion yet and this would require extra additional work to move STS client into CA, so let's wait until we can properly land this removal.

**Review by zirain — APPROVED** [review]

File: pilot/cmd/pilot-agent/options/security.go
Comment (kyessenov) [inline pilot/cmd/pilot-agent/options/security.go:133]: Is `CredFetcher` used anywhere else besides STS?

**Pull request closed without merging**
